contracts/EventBasedPredictionMarket.sol

```diff
@@ -0,0 +1,325 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "@uma/core/contracts/common/implementation/ExpandedERC20.sol";
+import "@uma/core/contracts/common/implementation/FixedPoint.sol";
+import "@uma/core/contracts/common/implementation/Testable.sol";
+import "@uma/core/contracts/oracle/implementation/Constants.sol";
+
+import "@uma/core/contracts/oracle/interfaces/FinderInterface.sol";
+import "@uma/core/contracts/oracle/interfaces/OptimisticOracleInterface.sol";
+
+// TODO use OptimisticOracleInterface from @uma/core once it's updated with setEventBased
+interface OptimisticOracleInterfaceEventBased {
+    function setEventBased(
+        bytes32 identifier,
+        uint256 timestamp,
+        bytes memory ancillaryData
+    ) external;
+}
+
+contract EventBasedPredictionMarket is Testable {
+    using FixedPoint for FixedPoint.Unsigned;
+    using SafeERC20 for ExpandedERC20;
+
+    /***************************************************
+     *  EVENT BASED PREDICTION MARKET DATA STRUCTURES  *
+     ***************************************************/
+    // Constants
+    uint256 ONE_ETHER = 1 ether;
```

Comment:
probs can just delete this and use `1e18` or `1 ether` where you use it later down the line.